src/compiler/printer.ts

```diff
@@ -528,12 +528,16 @@ const _super = (function (geti, seti) {
                     // JSX (non-expression)

```

Comment:
JsxExpression is not an expression, you can't put it in a CommaExpression for example. Rather its more like ExpressionStatement, it represents a different part of the tree that contains an expression. The fact that it inherits from `Expression` is a red-herring and one of the reasons I wrote this incorrectly in the first place.